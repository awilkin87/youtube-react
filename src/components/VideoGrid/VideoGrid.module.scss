.title {
  font-size: 1.2rem;
  margin-top: 24px;
}

.videoGrid {
  display: flex;
  flex-wrap: wrap;
  column-gap: 4px;
  
  & > * {
    margin-bottom: 24px;
  }
}
