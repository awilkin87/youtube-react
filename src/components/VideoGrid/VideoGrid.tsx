import { Divider } from 'semantic-ui-react';
import VideoPreview from '../VideoPreview/VideoPreview'
import styles from './VideoGrid.module.scss';

export interface VideoGridProps {
  title: string
  hideDivider?: boolean
}

/** Collection of videos, layed out in a grid, that are somehow related. */
const VideoGrid: React.FunctionComponent<Readonly<VideoGridProps>> = function (props) {
  return (
    <>
      <h4 className={styles.title}>{props.title}</h4>
      <div className={styles.videoGrid}>
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
      </div>
      { props.hideDivider || <Divider /> }
    </>
  );
}

export default VideoGrid;
