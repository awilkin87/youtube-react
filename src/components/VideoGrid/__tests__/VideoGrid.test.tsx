import { shallow } from "enzyme";
import VideoGrid from '../VideoGrid';

describe('VideoGrid', () => {
  test('Renders VideoGrid w/ divider', () => {
    const wrapper = shallow(<VideoGrid title='Grid Title' />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Renders VideoGrid w/o divider', () => {
    const wrapper = shallow(<VideoGrid hideDivider title='Grid Title' />);
    expect(wrapper).toMatchSnapshot();
  });
});
