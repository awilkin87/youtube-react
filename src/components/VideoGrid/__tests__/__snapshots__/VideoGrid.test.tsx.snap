// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid Renders VideoGrid w/ divider 1`] = `
<Fragment>
  <h4
    className="title"
  >
    Grid Title
  </h4>
   
  <div
    className="videoGrid"
  >
    <VideoPreview
      key="0"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="1"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="2"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="3"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="4"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="5"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="6"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="7"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="8"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="9"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="10"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="11"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid Renders VideoGrid w/o divider 1`] = `
<Fragment>
  <h4
    className="title"
  >
    Grid Title
  </h4>
   
  <div
    className="videoGrid"
  >
    <VideoPreview
      key="0"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="1"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="2"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="3"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="4"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="5"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="6"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="7"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="8"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="9"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="10"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
    <VideoPreview
      key="11"
      orientation="vertical"
      video={
        Object {
          "channelTitle": "Channel Name",
          "duration": "PT3M52S",
          "id": 0,
          "publishDate": 1970-01-01T00:00:00.000Z,
          "thumbnail": "http://via.placeholder.com/210x118",
          "videoTitle": "Video Title",
          "viewCount": 56,
        }
      }
    />
  </div>
  <Component />
</Fragment>
`;
